### DISCUSSION TITLE
Is there a technique to get a snapshot of the celestial sphere in vectors?

### DISCUSSION THREAD

**Discussion by ralienpp** [Stellarium/stellarium#3846] [Q&A]
Hi everyone, I'm authoring [an open source book of poetry for children](https://github.com/ralienpp/book-one). The text is in Russian, though some works are also translated to English.

Certain events occur at night, so a story may reference the sky and the stars, which are also featured in the illustrations. I used Stellarium to ensure the book has realistic depictions of what one would see if they looked at the sky from a given location (rather than random dots on a dark background which may be artistically okay, but scientifically inaccurate).

This is good enough for viewing the text on-screen, but if one wants to print this as an actual book, a higher resolution image is needed. Here is an example of what I have: https://github.com/ralienpp/book-one/blob/main/images/sahara-stars-annotated.png (screenshot of Stellarium, with some post processing in a graphics editor).

And this is where it is integrated into the text: https://github.com/ralienpp/book-one/blob/6ff7f980cfd5512eb5bf80d8ef53594469a210b3/main.tex#L1167, https://github.com/ralienpp/book-one/blob/6ff7f980cfd5512eb5bf80d8ef53594469a210b3/main.tex#L1467

Is there a way to get the data our in a vector graphics format? I'm specifically interested in the stars, though names and constellation lines would be a plus (otherwise I'd have to trace them by hand). If these details cannot be extracted easily out of Stellarium, perhaps you can recommend some other tool or technique?

I've found various web-sites that provide celestial maps in SVG, but they're not quite what I need. What I need is not a generic map of the stars in a given hemisphere, but a view tailored to a specific moment in time and location on the planet. So, if I were located at latX,lonY and looked at the celestial sphere in a given direction - this is what I'd see.

As a non-expert in the field, perhaps I've misused some terminology or am unaware of the right keywords to use to find what I need. I hope you can guide me in this process, thanks!

**Comment by alex-w** [comment]
Увы, Stellarium не умеет экспортировать снимки экрана в формате SVG - как вариант использовать PNG и выставить высокое разрешение для файлов.

**Reply by 10110111** [reply]
Пример того что есть сделан с помощью мобильной или веб-версии (стиль линий выдаёт), так что скриншот большего разрешения проблематично сделать.

Можно, однако, установить десктопную версию Stellarium, и тогда можно указать гораздо большее разрешение для скриншота (используя внутренние средства программы, а не клавишу <kbd>PrtScr</kbd>). Плюс можно будет выбрать язык (<kbd>F2</kbd> → *Язык культуры неба* для названий, *Язык программы* для сторон света), убрать свечение с горизонта (световое загрязнение).

Надо ещё заметить, что имеющаяся картинка подпорчена артефактами JPEG-компрессии. При создании и редактировании скриншота следует избегать промежуточных JPEG-файлов, сохраняя всё в PNG (или в файлы проекта редактора).

**Comment by gzotti** [comment]
Stellarium uses OpenGL, and this is a raster engine by definition. https://starplot.dev/ may be one place to create vector maps for configurable locations. 

It looks as if you had switched off atmosphere, but show fog and no ground in your image. This looks weird to me.

**Reply by 10110111** [reply]
OP's image seems to be the one from Stellarium Web, with ground (and maybe atmosphere) turned off. Notice how the constellation lines are like there (not touching the stars) rather than like in the desktop version.

**Reply by gzotti** [reply]
Ah, right! I had assumed he had worked that over a bit, but S-Web is the simpler explanation.

**Comment by ralienpp** [comment]
Hey everyone, thanks for all the feedback and your quick reactions, I greatly appreciate it. I reply in English, so that a wider audience can follow the thread.

Yes, I used Stellarium Web to produce the screenshot; I was unaware of the desktop version, but I tried it last night to get a higher resolution starting point. The technique I settled on:

- Start with such a screenshot, where all the labels and effects are disabled.
- Use the "trace bitmap" feature of Inkscape to transform the raster image into vectors.
- One needs to fiddle with the settings to get a crisp result.
- This gives you stars that are in the correct positions and have the right relative sizes.
- Then I _manually_ label the stars and trace the constellation lines (at a later stage).

Note that while the results are vectors, they're not optimal. What could have been a `<circle>` SVG element is actually a curve with redundant nodes, and is not perfectly round. 
![image](https://github.com/user-attachments/assets/c4120f3c-7639-43cf-83bd-6d389a9b55d2)
![image](https://github.com/user-attachments/assets/c935d3f4-692b-4620-875f-3f29c59dcff3)

If you want the Milky Way in there too, it can be done with an SVG filter.  In Inkscape, make a rectangular shape and bend it to roughly follow the galaxy, then fiddle with `Filters\Textures\Watercolor` until you're happy.

You end up with something like this:
![image](https://github.com/user-attachments/assets/a5c726d0-1882-432b-8936-d475ca7fb56d)


As gzotti pointed out, there is no ground - its silhouette needs to be added later (otherwise the bitmap tracing will give you an over-complicated shape). 



p.s. Following your suggestion, I also tried `Starplot`. There the approach is different and it places a greater emphasis on programming skills, rather than artistic skills. Stellarium is much easier to tinker with - the GUI makes the feedback loop very quick and you can rapidly iterate until you get what you want. With Starplot I had to understand what is declination and right ascension before I could "look" in the part of the sky I was interested in; but I still need to read more about optics to understand how to "zoom out" enough to see everything I want in the `OpticPlot`. Starplot produces elegant SVGs right out of the box, that's a major plus. Ultimately, I think I'll feature both types of visualizations in the book; this stuff is fascinating and hopefully it will inspire some kids to delve into astronomy.

**Reply by gzotti** [reply]
Starplot was just the first vector program I found that makes "dated" sky maps (with horizon). PP3 was used for the constellation maps in Wikipedia, and makes vectorized output (including Milky Way) mostly for LaTeX. fchart3 also looks great but seems to be for DSO finder charts only. You could find your scene in Stellarium, then take the relevant numbers for the Starplot configuration. Not sure why you need vector data in the first place, though.
